ArjunaJTS/jts/classes/com/arjuna/ats/internal/jts/orbspecific/interposition/coordinator/ServerTransaction.java

```diff
@@ -234,6 +235,16 @@ public final int doPhase2Commit ()
 
 		org.omg.CosTransactions.Status s = get_status();
 
+		if (s == org.omg.CosTransactions.Status.StatusCommitting || s == org.omg.CosTransactions.Status.StatusCommitted) {
+		    // This can move the entry if required
+		    TransactionCache.ReplayPhaseReturnStatus replayStatus = TransactionCache.replayPhase2(get_uid(), type());
```

Comment:
I need to look some more at this. I was hoping this TransactionCache call could be kept to the XATerminator called classes, i.e.: https://github.com/jbosstm/narayana/blob/master/ArjunaJTS/jtax/classes/com/arjuna/ats/internal/jta/transaction/jts/subordinate/jca/TransactionImple.java